shopfloor/actions/stock.py

```diff
@@ -121,21 +127,26 @@ def mark_move_line_as_picked(
         for line in move_lines:
             qty_done = quantity if quantity is not None else line.reserved_uom_qty
             line.qty_done = qty_done
-            line._split_partial_quantity()
+            if split:
+                line._split_partial_quantity()
             data = {
                 "shopfloor_user_id": user.id,
             }
             if package:
                 # destination package is set to the scanned one
                 data["result_package_id"] = package.id
             line.write(data)
+        values_assigned = {
+            "user_id": user.id,
+            "printed": True,
+        }
         # Extract the picked quantity in a split order and set current user
-        move_lines._extract_in_split_order(
-            {
-                "user_id": user.id,
-                "printed": True,
-            }
-        )
+        if split:
+            move_lines._extract_in_split_order(values_assigned)
+        else:
+            move_lines.picking_id.filtered(
+                lambda picking: not picking.printed or not picking.user_id == user
+            ).write(values_assigned)
```

Comment:
@rousseldenis +You must avoid the call to `write` on an empty recordset. Even if your recordset is empty it will cost time and ressource.